tests/disas/gc/struct-new.wat

```diff
@@ -0,0 +1,119 @@
+;;! target = "x86_64"
+;;! flags = "-W function-references,gc"
+;;! test = "optimize"
+
+(module
+  (type $ty (struct (field (mut f32))
+                    (field (mut i8))
+                    (field (mut anyref))))
+
+  (func (param f32 i32 anyref) (result (ref $ty))
+    (struct.new $ty (local.get 0) (local.get 1) (local.get 2))
+  )
+)
+;; function u0:0(i64 vmctx, i64, f32, i32, i32) -> i32 tail {
+;;     ss0 = explicit_slot 4, align = 4
+;;     gv0 = vmctx
+;;     gv1 = load.i64 notrap aligned readonly gv0+8
+;;     gv2 = load.i64 notrap aligned gv1
+;;     gv3 = vmctx
+;;     sig0 = (i64 vmctx, i32 uext, i32 uext, i32 uext, i32 uext) -> i32 system_v
+;;     fn0 = colocated u1:27 sig0
+;;     stack_limit = gv2
+;;
+;;                                 block0(v0: i64, v1: i64, v2: f32, v3: i32, v4: i32):
+;;                                     v71 = stack_addr.i64 ss0
+;;                                     store notrap v4, v71
+;; @002a                               v7 = iconst.i32 -1476395008
+;; @002a                               v8 = iconst.i32 0
+;; @002a                               v9 = iconst.i32 32
+;; @002a                               v10 = iconst.i32 8
+;; @002a                               v11 = call fn0(v0, v7, v8, v9, v10), stack_map=[i32 @ ss0+0]  ; v7 = -1476395008, v8 = 0, v9 = 32, v10 = 8
+;; @002a                               v15 = uextend.i64 v11
+;; @002a                               v16 = iconst.i64 16
+;; @002a                               v17 = uadd_overflow_trap v15, v16, user65535  ; v16 = 16
```

Comment:
As a minor possible optimization (future PR) this should be pretty easy to optimize in the egraph pass to downgrade this to `iadd`. It's pretty obvious by construction that this can't overflow, along with many other patterns too.

Whether it's worth it I'm not sure. Optimizing bounds checks is naturally going to be much more involved than just this b/c we want to rely on guard pages exclusively in theory.

Reply:
Ideally we'd have a mid-end ISLE rule something like

```lisp
(rule (simplify (uadd_overflow_trap $i64 x y))
    (if-let $true (add_cannot_overflow $I64 x y))
    (iadd x y))
```

where `add_cannot_overflow` handles constants and `(uextend ...)`s and such.

But because `uadd_overflow_trap` has side effects, we can't write such a rule in the mid-end today.